### ISSUE TITLE
Close trunk automatically when far away not working, closes the trunk instantly after opening

### ISSUE THREAD

**Issue by XenSide** [Trsak/esx_inventoryhud_trunk#14]
As of the time of writing this issue, in the file esx_trunk-lc,lua at row 203 you are using 

> local vehicle = GetClosestVehicle(pos["x"], pos["y"], pos["z"], 2.0, 0, 70)

which most of the time doesn't return a valid vehicle, the native used (GetClosestVehicle) is considered obsolete and many advise not to use it.

Also, depending on the lenght of the vehicle the native might or might not throw 0 instead of the vehicle id.

A fix suggestion is using the function you wrote "VehicleInFront"
So basically at row 203: 

> local vehicle = VehicleInFront()
> if vehicle ~= nil and vehicle ~= 0 then

I tested this solution myself and works pretty much flawlessly.

**Comment by DJJeffP** [comment]
Ik gonna test this also. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by DJJeffP** [comment]
I am checking the code about this topic and indeed there is no VihicleInFront() in that sections.
The Police part has it though. I changed it as followed:

**Civilian part (where the -- indicates the wrong part):**
```
Citizen.CreateThread(
  function()
    while true do
      Wait(0)
      local pos = GetEntityCoords(GetPlayerPed(-1))
      if CloseToVehicle then
        -- local vehicle = GetClosestVehicle(pos["x"], pos["y"], pos["z"], 2.0, 0, 70)
        local vehicle = VehicleInFront()
        -- if DoesEntityExist(vehicle) then
          if vehicle ~= nil and vehicle ~= 0 then
          CloseToVehicle = true
        else
          CloseToVehicle = false
          lastOpen = false
          ESX.UI.Menu.CloseAll()
          SetVehicleDoorShut(lastVehicle, 5, false)
        end
      end
    end
  end
)
```

**Police part (that is good and don't need a change) at 118**
   ```
 if not Config.CheckOwnership or (Config.AllowPolice and PlayerData.job.name == "police") or (Config.CheckOwnership and myVeh) then
      if globalplate ~= nil or globalplate ~= "" or globalplate ~= " " then
        CloseToVehicle = true
        local vehFront = VehicleInFront()
        local x, y, z = table.unpack(GetEntityCoords(GetPlayerPed(-1), true))
        local closecar = GetClosestVehicle(x, y, z, 5.0, 0, 71)

        if vehFront > 0 and closecar ~= nil and GetPedInVehicleSeat(closecar, -1) ~= GetPlayerPed(-1) then
          lastVehicle = vehFront
```

The solution is working pretty good, but........ It would be nice that you only can enter from the back and not all around the car.

**Comment by XenSide** [comment]
The part of the code edited in my solution shouldn't affect opening at all, that's only what makes the trunk in the car Close.

**Comment by DJJeffP** [comment]
It was just a note for the script writer ;-) forgot to tagged him in my message so here he is  @Trsak 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Trsak** [comment]
Closing since I have released new reworked version.
